### PR TITLE
fix: <Dropdown> controlled multi fix option remove

### PR DESCRIPTION
**Pull request by SergeyRoyt**
https://monday.monday.com/boards/3532714909/pulses/5370505249

### CODE REVIEW

File: src/components/Dropdown/Dropdown.jsx [hunk 1]
```diff
@@ -205,13 +205,13 @@ const Dropdown = forwardRef(
...
         }
-        const newSelectedOptions = selected.filter(option => option.value !== optionValue);
+        const newSelectedOptions = selectedOptions.filter(option => option.value !== optionValue);
         if (customOnChange) {
           customOnChange(newSelectedOptions, e);
         }
         setSelected(newSelectedOptions);
       };
-    }, [customOnChange, customOnOptionRemove, selected, selectedOptionsMap]);
+    }, [customOnChange, customOnOptionRemove, selectedOptions, selectedOptionsMap]);
```
Comment (SergeyRoyt) [inline src/components/Dropdown/Dropdown.jsx:214]: Actual bugfix

File: src/components/Dropdown/Dropdown.jsx [hunk 2]
```diff
@@ -251,7 +251,7 @@ const Dropdown = forwardRef(
           }
 
           if (!isControlled) {
-            setSelected([...selected, selectedOption]);
+            setSelected([...selectedOptions, selectedOption]);
```
Comment (SergeyRoyt) [inline src/components/Dropdown/Dropdown.jsx:254]: Just refactoring to help avoid such issues

File: src/components/Dropdown/__stories__/dropdown.stories.mdx [hunk 3]
```diff
@@ -498,6 +493,58 @@ The Dropdown component supports multiple options selection in two different stat
 
 ## Use cases and examples
 
+### Controlled multiple dropdown - demo story for Omri
```
Comment (SergeyRoyt) [inline src/components/Dropdown/__stories__/dropdown.stories.mdx:496]: Demo story to be removed

File: src/components/Dropdown/__stories__/dropdown.stories.mdx [hunk 4]
```diff
@@ -1,25 +1,20 @@
...
 import { OptionRenderer } from "./OptionRenderer.js";
-import Flex from "../../Flex/Flex";
-import { StoryDescription, Link } from "vibe-storybook-components";
-import { Attach, CloseSmall, Email } from "../../Icon/Icons";
-import { DialogContentContainer, DropdownMenu, DropdownOption, DropdownSingleValue, Icon } from "../../index";
-import Dialog from "../../Dialog/Dialog";
-import Button from "../../Button/Button";
+import { Attach, Email } from "../../Icon/Icons";
+import { DialogContentContainer, Dialog, Button, Flex, Avatar, Label } from "../../index";
 import "./dropdown.stories.scss";
```
Comment (SergeyRoyt) [inline src/components/Dropdown/__stories__/dropdown.stories.mdx:18]: Just refactoring

File: src/components/Dropdown/__stories__/dropdown.stories.mdx (hunk 3 above)
Comment (SergeyRoyt) [reply src/components/Dropdown/__stories__/dropdown.stories.mdx:496]: Also added a task into backlog to add story with controlled state

**Review by laviomri — APPROVED** [review]

**Pull request merged**